proofs/proofview.ml

```diff
@@ -1079,6 +1093,10 @@ struct
     let (sigma, c) = f (Evd.reset_future_goals sigma) in
     let evs = Evd.future_goals sigma in
     let evkmain = Evd.principal_future_goal sigma in
+    (** Redo the effects of sigma in the env *)
+    let privates_csts = Evd.eval_side_effects sigma in
+    let sideff = Safe_typing.side_effects_of_private_constants privates_csts in
+    let env = add_side_effects sideff env in
```

Comment:
Yes, and no. It will just replace the definition of a global constant with the same entry (i.e., the identity).